core/src/main/java/org/elasticsearch/index/Index.java

```diff
@@ -31,21 +32,23 @@
 public class Index implements Streamable {
 
     private String name;
+    private String uuid;
 
     private Index() {
 
     }
 
-    public Index(String name) {
+    public Index(String name, String uuid) {
```

Comment:
sure - didn't want to make things controversial.  I'll make a PR to remove the name.intern()